src/server.c

```diff
@@ -1296,13 +1296,12 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
         if (server.aof_state == AOF_ON &&
             !hasActiveChildProcess() &&
             server.aof_rewrite_perc &&
-            server.aof_current_size > server.aof_rewrite_min_size &&
-            !aofRewriteLimited())
+            server.aof_current_size > server.aof_rewrite_min_size)
         {
             long long base = server.aof_rewrite_base_size ?
                 server.aof_rewrite_base_size : 1;
             long long growth = (server.aof_current_size*100/base) - 100;
-            if (growth >= server.aof_rewrite_perc) {
+            if (growth >= server.aof_rewrite_perc && !aofRewriteLimited()) {
```

Comment:
i don't understand what's the benefit of moving the call here?
to avoid prints in case `growth < server.aof_rewrite_perc`?

arguably, we won't get into a situation where we have any limit to print if `growth` is small.

Reply:
i guess this was the problem:

> When the grow percent is not reached, the limit time will continue to increase until it reaches 60 minutes

but why did the AOFRW fail? what triggered it?
arguably, if it failed, we should retry (without any conditions)..
although i do agree that there's no point in increasing the time limit if we're not going to attempt a rewrite after that (since the conditions after the time limit checks are gonna fail.
so i agree it seems logical that this is the last condition

Reply:
in my test cases, after some rewrite failed, it can be triggered by three methods:
1. `bgrewriteaof` command failed twice, while the grow percent still not reached 100%
2. auto aofwrite failed twice, then change `auto-aof-rewrite-percentage` to a bigger number
3. auto aofwrite failed twice, then change `auto-aof-rewrite-min-size` to a bigger number